test/suites/standard1/router.robot

```diff
@@ -192,3 +223,21 @@ Setup Hello MicroShift Pods In Multiple Namespaces
     Expose Hello MicroShift    ${NS_OWNERSHIP_2}
     Oc Expose    svc hello-microshift --hostname ${HOSTNAME} --path /${NS_OWNERSHIP_1} -n ${NS_OWNERSHIP_1}
     Oc Expose    svc hello-microshift --hostname ${HOSTNAME} --path /${NS_OWNERSHIP_2} -n ${NS_OWNERSHIP_2}
+
+Http Port Should Be Open
+    [Documentation]    Try to curl the router, disregard the response because we only care about connectivity.
+    [Arguments]    ${port}
+    ${result}=    Access Hello MicroShift    ${port}
+    Should Match Regexp    ${result}    HTTP.*503
+
+Https Port Should Be Open
+    [Documentation]    Try to curl the router, disregard the response because we only care about connectivity.
+    [Arguments]    ${port}
+    ${result}=    Access Hello MicroShift    ${port}    protocol=https
+    Should Match Regexp    ${result}    HTTP.*503
+
+Port Should Be Closed
+    [Documentation]    Try to curl the router and expect a failure when connecting.
+    [Arguments]    ${port}
+    Run Keyword And Expect Error    *
```

Comment:
Can we write this in a way that asserts the type of error accessing the port receives? It might be easier to do from python than RF.

Reply:
Did it with curl in the end because it re-uses parts of the router connection keywords. Doing it in a separate library meant having part in RF and part in python (plus, to mimick the --connect-to tests you have to rewrite some connect functions as they are not part of the urllib API).